.github/nightly_matrix.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+const BRANCHES = ['master', 'PHP-8.1', 'PHP-8.0'];
+
+function get_branch_commit_cache_file_path(): string {
+    return dirname(__DIR__) . '/branch-commit-cache.json';
+}
+
+function get_branch_matrix(array $branches) {
+    $result = array_map(function ($branch) {
+        $branch_key = strtoupper(str_replace('.', '', $branch));
+        return [
+            'name' => $branch_key,
+            'ref' => $branch,
+        ];
+    }, $branches);
+
+    return $result;
+}
+
+function get_branches(string $pr_commit_hash) {
+    if (strlen($pr_commit_hash) !== 0) {
+        return [['name' => 'PR', 'ref' => $pr_commit_hash]];
+    }
+
+    $branch_commit_cache_file = get_branch_commit_cache_file_path();
+    $branch_commit_map = [];
+    if (file_exists($branch_commit_cache_file)) {
+        //$branch_commit_map = json_decode(file_get_contents($branch_commit_cache_file), JSON_THROW_ON_ERROR);
+    }
+
+    $changed_branches = [];
+    foreach (BRANCHES as $branch) {
+        $previous_commit_hash = $branch_commit_map[$branch] ?? null;
+        $current_commit_hash = trim(shell_exec('git rev-parse origin/' . $branch));
+
+        if ($previous_commit_hash !== $current_commit_hash) {
+            $changed_branches[] = $branch;
+        }
+
+        $branch_commit_map[$branch] = $current_commit_hash;
+    }
+
+    file_put_contents($branch_commit_cache_file, json_encode($branch_commit_map));
+
+    return get_branch_matrix($changed_branches);
+}
+
+function get_asan_matrix(array $branches) {
+    $jobs = [];
+    foreach ($branches as $branch) {
+        $jobs[] = [
+            'name' => '_ASAN_UBSAN',
+            'branch' => $branch['ref'],
+            'debug' => true,
+            'zts' => true,
+            'configuration_parameters' => "CFLAGS='-fsanitize=undefined,address -DZEND_TRACK_ARENA_ALLOC' LDFLAGS='-fsanitize=undefined,address'",
+            'run_tests_parameters' => '--asan',
+        ];
+    }
+    return $jobs;
+}
+
+$trigger = $argv[1] ?? 'schedule';
+$attempt = (int) ($argv[2] ?? 1);
+$pr_commit_hash = $argv[3] ?? '';
+
+$discard_cache = ($trigger === 'schedule' && $attempt !== 1) || $trigger === 'workflow_dispatch';
+if (!$discard_cache) {
+    @unlink(get_branch_commit_cache_file_path());
+}
```

Comment:
I believe the condition is inverted.
```suggestion
if ($discard_cache) {
    @unlink(get_branch_commit_cache_file_path());
}
```

Reply:
Indeed, inverted the variable name and value but not the if condition